### PR TITLE
fix(funding-rate-proposer): Make sure funding rate proposal timestamps use current EVM time, not current real world time

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

When war games testing the FundingRateProposer, @mrice32 helped me to realize that the `requestTimestamp` was being set to the current clock time, which is technically in the future from the EVM's point of view, causing unexpected reverts. 

I've added a `test` mode option to the proposer bot so that unit tests against the Timer's artificial time can still be run. The default setting is `test=false` which uses the latest block time as the request timestamp.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [X]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/funding-rate-proposer/index.js [hunk 1]
```diff
@@ -20,7 +20,6 @@ const { getWeb3 } = require("@uma/common");
  * @notice Runs strategies that request and propose new funding rates for Perpetual contracts.
  * @param {Object} logger Module responsible for sending logs.
  * @param {Object} web3 web3.js instance with unlocked wallets used for all on-chain connections.
- * @param {String} perpetualAddress Contract address of the Perpetual Contract.
```
Comment (nicholaspai) [inline packages/funding-rate-proposer/index.js:23]: Unused variable in this bot, perp addresses auto detected by querying PerpFactory past events

File: packages/funding-rate-proposer/src/proposer.js [hunk 2]
```diff
@@ -98,7 +98,7 @@ class FundingRateProposer {
     ]);
   }
 
-  async updateFundingRates() {
+  async updateFundingRates(useRealTime = false) {
```
Comment (mrice32) [inline packages/funding-rate-proposer/src/proposer.js:101]: Might be worth a note here about what environments useRealTime is set to true for.

**Review by mrice32 — APPROVED** [review]
LGTM!

**Review by chrismaree — APPROVED** [review]
great catch!

**Comment by Gamaranto** [comment]
Missing DCO then gtg @nicholaspai

**Pull request merged**